modules/asset/animal/src/Plugin/Asset/AssetType/Animal.php

```diff
@@ -43,10 +43,10 @@ public function buildFieldDefinitions() {
           'view' => -35,
         ],
       ],
-      'is_castrated' => [
+      'is_sterile' => [
         'type' => 'boolean',
-        'label' => $this->t('Castrated'),
-        'description' => $this->t('Has this animal been castrated?'),
+        'label' => $this->t('Sterile'),
+        'description' => $this->t('Is this animal sterile?'),
```

Comment:
Does/Should this support an `undefined` state where the sterility of the animal is unknown?

Reply:
I think you're right about deferring this. Though I do think it's a can of worms we'll need to get into...

Representing an explicit unknown state seems like it would be helpful for intake/processing scenarios. e.g. A animal shelter that receives a truckload of abandoned dogs - one probably wouldn't want to assume they're either already sterilized or not before going through and confirming on an animal by animal basis.

The sex of animals/plants (rabbits, ducks, hops, persimmon, ginko, etc) is another case where it could be important to be able to explicitly model the attribute as unknown and then also model *when* it becomes known.

Reply:
Yea those are good examples.